src/openms/source/CHEMISTRY/TheoreticalSpectrumGeneratorXLMS.cpp

```diff
@@ -574,15 +595,19 @@ namespace OpenMS
   void TheoreticalSpectrumGeneratorXLMS::addPrecursorPeaks_(PeakSpectrum & spectrum, DataArrays::IntegerDataArray & charges, DataArrays::StringDataArray & ion_names, double precursor_mass, int charge) const
   {
     Peak1D p;
-    String ion_name("[M+H]");
+    String ion_name;
 
     // precursor peak
     double mono_pos = precursor_mass + (Constants::PROTON_MASS_U * static_cast<double>(charge));
     p.setMZ(mono_pos / static_cast<double>(charge));
     p.setIntensity(pre_int_);
     if (add_metainfo_)
     {
+      ion_name = "[M+H]";
```

Comment:
do you need to copy the ion name into a variable before pushing it into ion_names? Otherwise, 
ion_names.emplace_back("[M+H]") could be slightly faster.

Reply:
I need to check where I can use that. I think in some cases I build up the strings from multiple components and it might not work there as easily, but for precursor peaks like this one it should work.
I will clean up the class a bit and also implement the faster neutral loss generation into this algorithm, but right now I only use this slower spectrum generation for the top X hits per spectrum. With my standard setting of top 5, this takes less than 0.5% of the total tool runtime on a small test example and will be even less with bigger protein databases.

Reply:
Ok it's not that important right now.